include/circt/Dialect/ESI/ESIChannels.td

```diff
@@ -197,6 +197,78 @@ def UnwrapSVInterfaceOp : ESI_Op<"unwrap.iface", [
   let hasVerifier = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// Channel bundles
+//===----------------------------------------------------------------------===//
+
+// Used to indicate the presence of a keyword in asm. Should be used within
+// optional parens in the declarative assembly format.
+// Upstreaming: https://github.com/llvm/llvm-project/pull/65438
+def UnitParameter : AttrOrTypeParameter<"::mlir::UnitAttr", "boolean flag"> {
+  let printer = "";
+  let parser = "::mlir::UnitAttr::get($_parser.getContext())";
+  let defaultValue = "::mlir::UnitAttr()";
+}
+
+def ChannelDirection : I32EnumAttr<"ChannelDirection",
+  "Direction of channel", [
```

Comment:
... relative to what? should probably add "see `ChannelBundleImpl` for details".